File: packet_seaquence/log/latest/work/src/protocol_analyzer/http.py
from src.protocol_analyzer.base import ProtocolAnalyzer

class AnalyzeHttp(ProtocolAnalyzer):
    """
    HTTP内の詳細情報を解析するクラス
    """

    def analyze(self):
        if not hasattr(self.packet, 'http'):
            return None
        try:
            http_info = {}
            if hasattr(self.packet.http, 'request_line'):
                http_info['request_line'] = self.packet.http.request_line
            if hasattr(self.packet.http, 'request_method'):
                http_info['method'] = self.packet.http.request_method
            if hasattr(self.packet.http, 'host'):
                http_info['host'] = self.packet.http.host
            if hasattr(self.packet.http, 'request_uri'):
                http_info['uri'] = self.packet.http.request_uri
            return http_info
        except Exception as e:
            print(f"HTTP解析中にエラーが発生しました: {str(e)}")
            return None

    def get_display_info(self):
        info = self.analyze()
        if not info:
            return "HTTP情報なし"
        line = info.get('request_line', '')
        method = info.get('method', '')
        uri = info.get('uri', '')
        if method and uri:
            return f"HTTP {method} {uri}"
        if line:
            return f"HTTP {line}"
        return "HTTP"

    def get_summary(self):
        info = self.analyze()
        if not info:
            return "HTTP"
        return f"HTTP {info.get('method','?')} {info.get('uri','?')}"

File: packet_seaquence/log/go_src/models/protocol_analyzer/http.go
// Package protocol_analyzer provides HTTP protocol analyzer
package protocol_analyzer

import (
	"fmt"
)

// AnalyzeHTTP implements the ProtocolAnalyzer interface for HTTP
type AnalyzeHTTP struct {
	BaseProtocolAnalyzer
}

// NewAnalyzeHTTP creates a new HTTP analyzer
func NewAnalyzeHTTP(packet Packet) *AnalyzeHTTP {
	return &AnalyzeHTTP{
		BaseProtocolAnalyzer: NewBaseProtocolAnalyzer(packet),
	}
}

// Analyze extracts HTTP information from a packet
func (a *AnalyzeHTTP) Analyze() map[string]interface{} {
	if !a.Pkt.LayerExists("http") {
		return nil
	}

	httpInfo := make(map[string]interface{})
	
	// Request line
	if requestLine, ok := a.Pkt.GetField("http", "request_line"); ok {
		httpInfo["request_line"] = requestLine
	}
	
	// Request method (GET, POST, etc.)
	if method, ok := a.Pkt.GetField("http", "request_method"); ok {
		httpInfo["method"] = method
	}
	
	// Host
	if host, ok := a.Pkt.GetField("http", "host"); ok {
		httpInfo["host"] = host
	}
	
	// Request URI
	if uri, ok := a.Pkt.GetField("http", "request_uri"); ok {
		httpInfo["uri"] = uri
	}
	
	return httpInfo
}

// GetDisplayInfo returns formatted HTTP information
func (a *AnalyzeHTTP) GetDisplayInfo() string {
	info := a.Analyze()
	if info == nil {
		return "HTTP情報なし"
	}
	
	line := getOrDefault(info["request_line"], "")
	method := getOrDefault(info["method"], "")
	uri := getOrDefault(info["uri"], "")
	
	if method != "" && uri != "" {
		return fmt.Sprintf("HTTP %s %s", method, uri)
	}
	
	if line != "" {
		return fmt.Sprintf("HTTP %s", line)
	}
	
	return "HTTP"
}

// GetSummary returns a summary of HTTP information
func (a *AnalyzeHTTP) GetSummary() string {
	info := a.Analyze()
	if info == nil {
		return "HTTP"
	}
	
	method := getOrDefault(info["method"], "?")
	uri := getOrDefault(info["uri"], "?")
	
	return fmt.Sprintf("HTTP %s %s", method, uri)
}